torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -202,6 +204,21 @@ def _check_flat_params_on_expected_device(state: _FSDPState, module: nn.Module):
             )
 
 
+import torch.distributed as dist
+
+
+def _init_device_mesh(
+    pg: ProcessGroup,
+) -> Optional[DeviceMesh]:
+    # We are testing for dist.get_world_size(pg) == dist.get_world_size() for now.
+    # TODO: Address cases when dist.get_world_size(pg) != dist.get_world_size().
+    if dist.get_world_size(pg) != dist.get_world_size():
+        return None
```

Comment:
Should we raise an error here? It seems this would be a silent failure downstream?

Reply:
> Should we raise an error here? It seems this would be a silent failure downstream?

Hm. I was thinking we don't want to affect the existing workflow. 

The subsequent PRs will be using the device_mesh to create dtensor state_dict. I will be adding a flag(use_dtensor set to False by default) for the state dict config to control that. For existing workflow, they might or might not have a _device_mesh in _FSDPState, but _device_mesh won't be used downstream at all. Hence, just returning None for now. 

In later PR, we can raise an error if use_dtensor is set to True but _FSDPState._device_mesh = None. Wdyt?